miv/io/data.py

```diff
@@ -379,3 +520,106 @@ def insert(self, idx, data):
             self.data_list.insert(idx, data)
         else:
             logging.warning("Invalid data cannot be loaded to the DataManager.")
+
+    def auto_channel_mask_baseline(
+        self,
+        filter: FilterProtocol,
+        detector: SpikeDetectionProtocol,
+        no_spike_threshold: float = 1,
+    ):
+        """
+        Perform automatic channel masking.
+        This method simply applies a Butterworth filter, extract spikes, and filter out
+        the channels that contain either no spikes or too many spikes.
+
+        Parameters
+        ----------
+        filter : FilterProtocol
+            Filter that is applied to the signals before detecting spikes.
+        detector : SpikeDetectionProtocol
+            Spike detector that is used to extract spikes from the filtered signal.
+        no_spike_threshold : float, default=1
+            Spike rate threshold (spike per sec) for filtering channels with no spikes.
+            (default = 1)
+
+        """
+
+        for data in self.data_list:
+            with data.load() as (sig, times, samp):
+                mask_list = []
+
+                filtered_signal = filter(sig, samp)
+                spiketrains = detector(filtered_signal, times, samp)
+                spike_stats = firing_rates(spiketrains)
+
+                for channel in range(len(spike_stats["rates"])):
+                    if spike_stats["rates"][channel] <= no_spike_threshold:
+                        mask_list.append(channel)
+
+                data.set_channel_mask(mask_list)
+
+    def auto_channel_mask(
+        self,
+        spontaneous_data: Data,
+        filter: FilterProtocol,
+        detector: SpikeDetectionProtocol,
+        omit_experiments: Iterable[int] = [],
```

Comment:
It is not a good practice to initialize an empty list. An empty list is a mutable object in python, and it is not a good practice to use it as an initializer. Try to pass None and have an if statement to convert `None` to list.
```suggestion
        omit_experiments: Optional[Iterable[int]] = None,
```

Reply:
7246ca6